caikit_nlp/modules/text_classification/sequence_classification.py

```diff
@@ -0,0 +1,235 @@
+from caikit.core.toolkit import error_handler
+import alog
+
+# Local
+from ...data_model import Classification, ClassificationResult
+from .text_classification_task import TextClassificationTask
+
+log = alog.use_channel("SEQ_CLASS")
+error = error_handler.get(log)
+
+
+@module(
+    id="d21107ca-d579-4321-aedd-3099a526e0dd",
+    name="Sequence classification",
+    version="0.1.0",
+    task=TextClassificationTask,
+)
+class SequenceClassification(ModuleBase):
+
+    ################################ Constructor #################################################
+
+    def __init__(
+        self,
+        tokenizer: PreTrainedTokenizerBase,
+        model: PreTrainedModel,
+        device: Union[str, int, None],
+    ):
+        super().__init__()
+        error.type_check(
+            "<NLP74125820E>",
+            PreTrainedTokenizerBase,
+            tokenizer=tokenizer,
+        )
+        error.type_check(
+            "<NLP64751996E>",
+            PreTrainedModel,
+            model=model,
+        )
+        self.tokenizer = tokenizer
+        self.model = model
+        self.device = device
+
+    ################################## API functions #############################################
+
+    def run(self, text: str) -> ClassificationResult:
+        """Run the sequence classification, truncates sequences too long for model
+
+        Args:
+            text: str
+                Input string to be classified
+
+        Returns:
+            ClassificationResult
+        """
+        scores_dict = self._get_scores(text)
+        # Re-organize scores_dict - for one text, this is just the first score
+        return SequenceClassification._process_predictions(scores_dict, text_idx=0)
+
+    def run_batch(self, texts: List[str]) -> List[ClassificationResult]:
+        """Run the sequence classification on batch, truncates sequences too long for model
+
+        Args:
+            text: List[str]
+                Input strings to be classified
+
+        Returns:
+            List[ClassificationResult]
+        """
+        scores_dict = self._get_scores(texts)
+        num_texts = len(texts)
+
+        # Re-organize scores_dict for each example
+        # We could eventually consider whether or not to sort classifications by scores
+        # but avoiding this prescription here for now
+        classification_predictions = []
+        for text_idx in range(num_texts):
+            classification_prediction = SequenceClassification._process_predictions(
+                scores_dict, text_idx
+            )
+            classification_predictions.append(classification_prediction)
+        return classification_predictions
+
+    def save(self, model_path: str):
+        """Save model in target path
+
+        Args:
+            model_path: str
+                Path to store model artifact(s)
+        """
+        self.tokenizer.save_pretrained(model_path)
+        self.model.save_pretrained(model_path)
```

Comment:
I meant this to be a very thin layer around HF `SequenceClassification` models - this had been discussed a bit offline in the sense that with `ModuleConfig`, this would involve adding another layer of config around being able to work with Huggingface models

Reply:
Discussed offline - switched to `ModuleConfig` paradigm assuming future accommodation for HF/transformers models directly that will not always require nesting. This `save` logic is now done via pretrained model resource